### ISSUE TITLE
Dir.mktmpdir errors on apple-silicon native JDK

### ISSUE THREAD

**Issue by fidothe** [jruby/jruby#6974]
**Environment Information**
 
* JRuby 9.3.2.0 (also tried 9.3.1.0). Repro works with plain jruby -e, no flags, `JRUBY_OPTS` is unset.
* macOS 12.1, zulu-11.jdk 11.0.13, zulu-17.jdk 17.0.1

**Expected Behavior**

(under MRI)

```shell
$ ruby -e "require 'tmpdir'; Dir.mktmpdir { |d| puts d }"
/var/folders/h0/14gvbwnx11b41_pds3qld_y40000gn/T/d20219217-34690-mo7zzw
```

**Actual Behavior**

```shell
$ jruby -e "require 'tmpdir'; Dir.mktmpdir { |d| puts d }"
ArgumentError: could not find a temporary directory
    tmpdir at /Users/matt/.rubies/jruby-9.3.2.0/lib/ruby/stdlib/tmpdir.rb:35
    tmpdir at /Users/matt/.rubies/jruby-9.3.2.0/lib/ruby/stdlib/tmpdir.rb:112
    create at /Users/matt/.rubies/jruby-9.3.2.0/lib/ruby/stdlib/tmpdir.rb:120
  mktmpdir at /Users/matt/.rubies/jruby-9.3.2.0/lib/ruby/stdlib/tmpdir.rb:87
    <main> at -e:1
```

I'm guessing this is more code signing / permissions stuff...

**Comment by 545ch4** [comment]
Same here.

At `lib/ruby/stdlib/tmpdir.rb` it is expected, that a created tmpdir should be a directory, while at apple-silicon jruby it is not... To reproduce:

```
$ mkdir /tmp/test
$ RBENV_VERSION=2.7.5 ruby -e "puts File.lstat('/tmp').inspect; puts File.lstat('/tmp').directory?; puts File.lstat('/tmp/test').inspect; puts File.lstat('/tmp/test').directory?"
#<File::Stat dev=0x1000011, ino=1152921500312809861, mode=0120755, nlink=1, uid=0, gid=0, rdev=0x0, size=11, blksize=4096, blocks=0, atime=2021-12-08 00:39:27 +0100, mtime=2021-12-08 00:39:27 +0100, ctime=2021-12-08 00:39:27 +0100, birthtime=2021-12-08 00:39:27 +0100>
false
#<File::Stat dev=0x1000011, ino=3196862, mode=040755, nlink=2, uid=501, gid=0, rdev=0x0, size=64, blksize=4096, blocks=0, atime=2021-12-28 14:25:01.768275781 +0100, mtime=2021-12-28 14:25:01.768275781 +0100, ctime=2021-12-28 14:25:01.768275781 +0100, birthtime=2021-12-28 14:25:01.768275781 +0100>
true
$ jruby -e "puts File.lstat('/tmp').inspect; puts File.lstat('/tmp').directory?; puts File.lstat('/tmp/test').inspect; puts File.lstat('/tmp/test').directory?"
#<File::Stat dev=0x1000011, ino=106989, mode=030605, nlink=14, uid=268435455, gid=0, rdev=0x0, size=0, blksize=0, blocks=1638920367, atime=1970-01-01 01:00:01 +0100, mtime=1970-01-01 01:00:01 +0100, ctime=1970-01-01 01:00:01 +0100, birthtime=2021-12-08 00:39:27 +0100>
false
#<File::Stat dev=0x1000011, ino=147949, mode=0143676, nlink=48, uid=0, gid=501, rdev=0x0, size=768275781, blksize=768275781, blocks=1640697901, atime=1970-01-01 01:00:01 +0100, mtime=1994-05-07 03:56:22 +0200, ctime=1994-05-07 03:56:22 +0200, birthtime=2021-12-28 14:25:01 +0100>
false
```

All jruby stat dates are looking weired, `uid` and `gid` are incorrect, and certainly `/tmp/test` **is** a directory.

Could be related/equal to https://github.com/jruby/jruby/issues/5521 => https://github.com/jnr/jnr-posix/issues/126

**Comment by headius** [comment]
Likely fixed by https://github.com/jnr/jnr-posix/pull/173 which has merged into all branches. Marking as fixed in 9.3, which is the release that will officially support Apple Silicon.
